domenetjenester/registerinnhenting/src/main/java/no/nav/ung/sak/domene/registerinnhenting/InntektAbonnentTjeneste.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Dependent
+public class InntektAbonnentTjeneste {
+
+    private static final String UNG_INNTEKT_FORMAAL = "Ung";
+    private static final String UNG_INNTEKT_FILTER = "Ung";
+    private static final int BEVARINGTID_I_INNTEKTSKOMPONENTEN_MAANEDER = 1;
+    private static final Logger log = LoggerFactory.getLogger(InntektAbonnentTjeneste.class);
+
+    private InntektAbonnementRepository inntektAbonnementRepository;
+    private InntektAbonnentKlient inntektAbonnentKlient;
+    private FagsakTjeneste fagsakTjeneste;
+    private TpsTjeneste tpsTjeneste;
+
+
+    public InntektAbonnentTjeneste() {
+        // for CDI proxy
+    }
+
+    @Inject
+    public InntektAbonnentTjeneste(InntektAbonnementRepository inntektAbonnementRepository,
+                                   InntektAbonnentKlient inntektAbonnentKlient,
+                                   FagsakTjeneste fagsakTjeneste, TpsTjeneste tpsTjeneste) {
+        this.inntektAbonnementRepository = inntektAbonnementRepository;
+        this.inntektAbonnentKlient = inntektAbonnentKlient;
+        this.fagsakTjeneste = fagsakTjeneste;
+        this.tpsTjeneste = tpsTjeneste;
+    }
+
+    public void opprettAbonnement(AktørId aktørId, Periode periode){
+        var eksisterendeAbonnement = inntektAbonnementRepository.hentAbonnementForAktør(aktørId).orElse(null);
+        if (eksisterendeAbonnement != null) {
+            if (eksisterendeAbonnement.getPeriode().tilPeriode().equals(periode)) {
+                log.info("Prøver å opprette abonnement for aktør, men abonnementID = {} eksisterer allerede for denne periode", eksisterendeAbonnement.getAbonnementId());
+                return;
+            } else {
+                throw new IllegalStateException("Prøver å opprette at abonnement, men det eksisterer en abbonnentId = {} for aktøren, på en annen periode" + eksisterendeAbonnement.getAbonnementId());
+            }
+        }
+
+        var tomFagsakPeriode = fagsakTjeneste.finnFagsakerForAktør(aktørId).stream()
+            .filter(Fagsak::erÅpen)
+            .map(fagsak -> fagsak.getPeriode().getTomDato())
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("Fant ingen åpen fagsak med gyldig periode"));
+
+        var personIdent = tpsTjeneste.hentFnr(aktørId).orElseThrow();
+        long abonnementId = inntektAbonnentKlient.opprettAbonnement(
+            personIdent,
+            UNG_INNTEKT_FORMAAL,
+            List.of(UNG_INNTEKT_FILTER),
+            YearMonth.from(periode.getFom()),
+            YearMonth.from(periode.getTom()),
+            tomFagsakPeriode,
+            BEVARINGTID_I_INNTEKTSKOMPONENTEN_MAANEDER
+        );
+        inntektAbonnementRepository.lagre(new InntektAbonnement(String.valueOf(abonnementId), aktørId));
+    }
+
+    public Optional<Long> hentFørsteSekvensnummer() {
+        return  inntektAbonnentKlient.hentStartSekvensnummer(LocalDate.now());
```

Comment:
Har du mulighet til å gjøre en reformattering i intellij? Her er det et mellomrom for mye, kanskje det er mer inkonsistent formattering i filene? Det blir enklere å lese om det er likt